lint/log_handler.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import sublime
+
+from . import util
+
+
+DEBUG_FALSE_LEVEL = logging.WARNING
+DEBUG_TRUE_LEVEL = logging.INFO
+ERROR_PANEL_LEVEL = logging.WARNING
+
+logger = logging.getLogger('SublimeLinter')
+logger.setLevel(logging.DEBUG)
+handler = None
+error_panel_handler = None
+
+
+def install():
+    install_std_handler()
+    install_error_panel_handler()
+
+    settings = sublime.load_settings("SublimeLinter.sublime-settings")
+    settings.add_on_change('SublimeLinter._logging', install_std_handler)
+
+
+def uninstall():
+    if handler:
+        logger.removeHandler(handler)
+    if error_panel_handler:
+        logger.removeHandler(error_panel_handler)
+
+    settings = sublime.load_settings("SublimeLinter.sublime-settings")
+    settings.clear_on_change('SublimeLinter._logging')
+
+
+def install_std_handler():
+    global handler
+    if handler:
+        logger.removeHandler(handler)
+
+    settings = sublime.load_settings("SublimeLinter.sublime-settings")
+    level = settings.get('debug', False)
+
+    if level is False:
+        level = DEBUG_FALSE_LEVEL
+        formatter = TaskNumberFormatter(
+            fmt="SublimeLinter: {LEVELNAME}{message}",
+            style='{')
+        handler = logging.StreamHandler()
+        handler.setFormatter(formatter)
+    else:
+        if level is True:
+            level = DEBUG_TRUE_LEVEL
+        else:
+            level = logging.getLevelName(level.upper())
```

Comment:
Shouldn't this be `getattr(logging, level.upper())`? `logging.getLevelName` is the reverse operation.

Reply:
Hm, I have it from here https://github.com/python/cpython/blob/504f19145ca5738162d6a720fa45b364ac8c0384/Lib/logging/__init__.py#L120-L141

Reply:
Yeah, but you already have a string in `level` and want the numeric level for `handler.setLevel`.

Reply:
You did not read the code, did you

Reply:
No, I actually did not. I only read the docstring and it looks like this behavior is undocumented (or rather a backwards compatibility thing that has been removed for 3.4) which is why this behavior confused me. And admittedly, for a function named `getLevelName` to return an integer with the level (and not its name) *is* pretty confusing.

Related upstream issue: https://bugs.python.org/issue22386